bundles/org.eclipse.osgi/felix/src/org/apache/felix/resolver/Candidates.java

```diff
@@ -1145,7 +1150,15 @@ public Candidates copy()
                 m_delta.deepClone());
     }
 
-    public void dump(ResolveContext rc)
+    /**
+     * Dump the current candidate set to system out
+     * 
+     * @param rc  the resolve context that should be used to look for existing
+     *            wirings
+     * @param all if true all requirements are printed, if false only those that
+     *            have more than one provider
+     */
+    public void dump(ResolveContext rc, boolean all, PrintStream printStream)
```

Comment:
What calls this?

Reply:
This is / was used in Felix for debugging the current state of the resolve context, I just generalized that so one can redirect the output also to a (file) printstream instead of system out.